src/utils/CometAdapter.cpp

```diff
@@ -0,0 +1,589 @@
+    //vector<String> valid_strings = ListUtils::create<String>("ppm,Da");
+    registerIntOption_("precursor_error_units", "<choice>", 2, "peptide_mass_units (Comet) 0=amu, 1=mmu, 2=ppm", false, false);
+    //registerIntOption_("mass_type_parent", "<num>", 1, "0=average masses, 1=monoisotopic masses", false, true);
+    //registerIntOption_("mass_type_fragment", "<num>", 1, "0=average masses, 1=monoisotopic masses", false, true);
+    //registerIntOption_("precursor_tolerance_type", "<num>", 0, "0=average masses, 1=monoisotopic masses", false, false);
+    registerIntOption_("isotope_error", "<num>", 0, "0=off, 1=on -1/0/1/2/3 (standard C13 error), 2= -8/-4/0/4/8 (for +4/+8 labeling)", false, false);
+
+    //Search Enzyme
+    vector<String> all_enzymes;
+    EnzymesDB::getInstance()->getAllCometNames(all_enzymes);
+    registerStringOption_("enzyme", "<cleavage site>", "Trypsin", "The enzyme used for peptide digestion.", false, false);
+    setValidStrings_("enzyme", all_enzymes);
+    // TO DO: change selection menue from numbers to valid strings
+    registerIntOption_("num_enzyme_termini", "<num>", 2, "1 (semi-digested), 2 (fully digested, default), 8 C-term unspecific , 9 N-term unspecific", false, false);
+    registerIntOption_("allowed_missed_cleavages", "<num>", 1, "Number of possible cleavage sites missed by the enzyme, maximum value is 5; for enzyme search", false, false);
+
+    //Fragment Ions
+    registerDoubleOption_("fragment_bin_tolerance", "<tolerance>", 1.0005, "fragment_mass_tolerance (MSGF+), fragment_bin_tol (Comet)", false, true);
+    registerDoubleOption_("fragment_bin_offset", "<tolerance>", 0.25, "fragment_bin_offset (Comet)", false, true);
+    registerIntOption_("theoretical_fragment_ions", "<num>", 0, "theoretical fragment ion peak representation, 0==sum of intensites plus flanking bins, 1==sum of intensities of central bin only", false, true);
+    registerIntOption_("use_A_ions", "<num>", 0, "use A ions for PSM, 0 == no, 1 == yes", false, true);
+    registerIntOption_("use_B_ions", "<num>", 1, "use B ions for PSM, 0 == no, 1 == yes", false, true);
+    registerIntOption_("use_C_ions", "<num>", 0, "use C ions for PSM, 0 == no, 1 == yes", false, true);
+    registerIntOption_("use_X_ions", "<num>", 0, "use X ions for PSM, 0 == no, 1 == yes", false, true);
+    registerIntOption_("use_Y_ions", "<num>", 1, "use Y ions for PSM, 0 == no, 1 == yes", false, true);
+    registerIntOption_("use_Z_ions", "<num>", 0, "use Z ions for PSM, 0 == no, 1 == yes", false, true);
+    registerIntOption_("use_NL_ions", "<num>", 0, "use Neutral Loss ions for PSM, 0 == no, 1 == yes", false, true);
+
+    //Output
+    registerIntOption_("num_hits", "<num>", 5, "Number of peptide hits in output file", false, false);
+
+    //mzXML/mzML parameters
+    registerStringOption_("precursor_charge", "[min] [max]", "0 0", "charge range to search: 0 0 == search all charges, 2 6 == from +2 to +6, 3 3 == +3", false, false);
```

Comment:
OpenMS generally uses the notation "[min]:[max]" for ranges. See `TOPPBase::parseRange_`.